File: Data Structures & Algorithms/count-paths/submission-0.cpp
class Solution {
public:


    int solve(int i , int j ){

        if(i <0 || j<0)return 0 ;
        if(i==0 && j==0) return 1;

    

        int up= solve(i-1 , j);
        int left= solve(i , j-1);
        
        return up + left;

    }

    int uniquePaths(int m, int n) {

        return solve(m-1 , n-1);
        
    }
};

File: Data Structures & Algorithms/capacity-to-ship-packages-within-d-days/submission-0.py
class Solution:
    def shipWithinDays(self, weights: List[int], days: int) -> int:

        left = max(weights)
        right = sum(weights)

        while left < right :
            mid = (left + right)//2
            current_load = 0
            required_days = 1

            for w in weights :
                if current_load + w > mid :
                    required_days +=1
                    current_load = w
                    if required_days>days:
                        break 
                else:
                    current_load += w
            if required_days <= days:
                right = mid 
            else :
                left = mid+1
         
        return left

                    

            
